logger/common.go

```diff
@@ -495,9 +490,13 @@ func SetUIDAndGID(uid int, gid int) error {
 }
 
 // setUID sets UID of current goroutine.
+// If you are building with go version includes the following commit, this syscall would apply
+// to current process, otherwise it would only apply to current goroutine.
+// Commit: https://github.com/golang/go/commit/d1b1145cace8b968307f9311ff611e4bb810710c
 func setUID(id int) error {
-	if _, _, errno := syscall.Syscall(syscall.SYS_SETUID, uintptr(id), 0, 0); errno != 0 {
```

Comment:
Are we changing from `syscall.Syscall()` to `syscall.Setuid()` because the fix to Golang only affected the 9 functions mentioned in https://github.com/golang/go/commit/d1b1145cace8b968307f9311ff611e4bb810710c and not `syscall.Syscall()`, or is the change primarily for readability/simplicity? Just curious if the already existing implementation works as well.

Reply:
The original implementation does not work as they only fixed the `syscall.Setuid()` function. They basically called a function in `syscall.Setuid()` to apply the syscall to all threads and that's why the original implementation is not working.